### PR TITLE
Do not throw exception on deprecated ignoredisk --interactive

### PR DESCRIPTION
**Pull request by rvykydal**
The option was deprecated by
commit 67bde6b1bfe0a20d17a95620d56df96dfc805efej
but we don't want to abort the installation for existing kickstarts using the
option so loose the guard in parse method and add a deprecation warning.

If there is the --interactive option only, just issue the warning, do not raise
parse error. (These kickstarts used to be accepted.)

If there is also other option present in the command, modify the message of
raised exception with information that the --interactive option is no longer
supported. (These kickstarts used to raise the exception but the message should
be changed.)

Resolves: rhbz#1637872

### CODE REVIEW

**Comment by rvykydal** [comment]
Note: we still don't have mechanism to pass warnings from modules (Storage module in this case) to anaconda/Boss so the message will be visible only in Storage module log and won't be exposed when using inst.ksstrict option.

File: pykickstart/commands/ignoredisk.py [hunk 1]
```diff
@@ -151,7 +153,29 @@ class F29_IgnoreDisk(F14_IgnoreDisk):
...
+        if self.onlyuse:
+            howmany += 1
+        if self.interactive:
+            howmany += 1
+            warnings.warn(_("Ignoring deprecated option on line %s:  The ignoredisk command "
+                            "no longer supports --interactive option.  In future releases, "
+                            "this will result in a fatal error from kickstart. "
+                            "Please modify your kickstart file to remove the option.")
+                          % self.lineno, KickstartDeprecationWarning)
+        if howmany != 1:
```
Comment (jkonecny12) [inline pykickstart/commands/ignoredisk.py]: What about removing the `howmany` and change this line to:

```suggestion
        if sum(map(int, (self.ignoredisk, self.onlyuse, self.interactive))) != 1:
```

File: pykickstart/commands/ignoredisk.py [hunk 2]
```diff
@@ -151,7 +153,29 @@ class F29_IgnoreDisk(F14_IgnoreDisk):
...
+        retval = FC3_IgnoreDisk.parse(self, args)
+
+        howmany = 0
+        if self.ignoredisk:
+            howmany += 1
+        if self.onlyuse:
+            howmany += 1
+        if self.interactive:
+            howmany += 1
+            warnings.warn(_("Ignoring deprecated option on line %s:  The ignoredisk command "
```
Comment (poncovka) [inline pykickstart/commands/ignoredisk.py]: I think that this warning should be already produced by the argument parser.

Comment (rvykydal) [reply pykickstart/commands/ignoredisk.py]: I am not able to find the warning anywhere in the logs. Maybe this holds only for deprecated commands?
BTW, the warning will be only displayed in storage module log as we don't forward kickstart parsing warnings from modules to anaconda (just errors). Which is perhaps ok for now.

File: pykickstart/commands/ignoredisk.py (hunk 1 above)
Comment (rvykydal) [reply pykickstart/commands/ignoredisk.py]: Yes, howmany seems a bit funny to me, but I prefer to keep it consistent with other code.

Comment (jkonecny12) [reply pykickstart/commands/ignoredisk.py]: OK, just an idea :). You know in deep of my hearth I like functional programming :D.

**Pull request merged**